Automattic/wp-calypso
# Plans: Remove CartTrialAd and FreeTrialNotice

As per conclusion of #9593. Since free trials have been disabled for about a year, this is dead code that we're dragging along.
This also means we can remove the `sitePlans` prop from a number of ancestor components.

To test:
Verify that Calypso still works, with special focus on Plans and Domains/Domain Search. I removed a number of props that apparently are now obsolete, as well as accompanying query components; however, there is a chance that descendant components were relying on data fetched by those query components and were broken by their removal. To make sure, clear the indexed db cache, and reload the page.

cc @gwwar @lamosty @scruffian
